backend/database-fs.js

```diff
@@ -36,38 +35,59 @@ async function testCaseSensitivity () {
   }
 }
 
-export async function initStorage (options: Object = {}): Promise<void> {
-  dataFolder = resolve(options.dirname)
-  await mkdir(dataFolder, { mode: 0o750, recursive: true })
-  await testCaseSensitivity()
-}
+export default class FsBackend extends DatabaseBackend {
+  dataFolder: string = ''
+  depth: number = 0
+  keyChunkLength: number = 2
 
-// Useful in test hooks.
-export function clear (): Promise<void> {
-  return readdir(dataFolder)
-    .then(keys => Promise.all(keys.map(key => unlink(join(dataFolder, key)))))
-}
+  constructor (options: Object = {}) {
+    super()
+    this.dataFolder = resolve(options.dirname)
+    if (options.depth) this.depth = options.depth
+    if (options.keyChunkLength) this.keyChunkLength = options.keyChunkLength
+  }
 
-// eslint-disable-next-line require-await
-export async function readData (key: string): Promise<Buffer | string | void> {
-  // Necessary here to thwart path traversal attacks.
-  checkKey(key)
-  return readFile(join(dataFolder, key))
-    .catch(err => undefined) // eslint-disable-line node/handle-callback-err
-}
+  // Maps a given key to a real path on the filesystem.
+  mapKey (key: string): string {
+    if (!this.depth) return join(this.dataFolder, key)
+    // TODO: optimize if necessary.
+    const keyChunks = key.match(new RegExp('[A-Za-z=]{1,' + this.keyChunkLength + '}', 'g')) ?? []
+    return join(this.dataFolder, ...keyChunks.slice(0, this.depth), keyChunks.slice(this.depth).join(''))
+  }
 
-// eslint-disable-next-line require-await
-export async function writeData (key: string, value: Buffer | string): Promise<void> {
-  return writeFile(join(dataFolder, key), value)
-}
+  async init () {
+    await mkdir(this.dataFolder, { mode: 0o750, recursive: true })
+    await testCaseSensitivity(this)
+  }
 
-// eslint-disable-next-line require-await
-export async function deleteData (key: string): Promise<void> {
-  return unlink(join(dataFolder, key)).catch(e => {
-    // Ignore 'not found' errors
-    if (e?.code === 'ENOENT') {
-      return
-    }
-    throw e
-  })
+  async clear () {
+    await Promise.all(
+      await readdir(this.dataFolder)
+        .then(names => names.map(name => join(this.dataFolder, name)))
+        .then(paths => paths.map(path => rm(path, { recursive: true })))
+    )
+  }
```

Comment:
`o3-mini-high` suggests making this code simpler/clearer, and I like its suggestion:

```js
async clear () {
  const names = await readdir(this.dataFolder)
  const paths = names.map(name => join(this.dataFolder, name))
  await Promise.all(paths.map(path => rm(path, { recursive: true })))
}
```